File: Sources/Conversations.swift
import Foundation

// ============================================================================
// Conversations.swift — varias conversaciones con el slime, persistidas.
// ============================================================================

struct Msg: Codable {
    var role: String           // "user" | "assistant"
    var content: String
    var imagePath: String? = nil   // captura adjunta (thumbnail clicable)
}

struct Conversation: Codable {
    var id: String
    var title: String
    var messages: [Msg]

    static func new() -> Conversation {
        Conversation(id: UUID().uuidString, title: "Nueva conversación", messages: [])
    }
}

struct ConversationStore: Codable {
    var conversations: [Conversation] = []

    static var fileURL: URL {
        let base = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask)[0]
            .appendingPathComponent("SlimePet", isDirectory: true)
        try? FileManager.default.createDirectory(at: base, withIntermediateDirectories: true)
        return base.appendingPathComponent("conversations.json")
    }

    static func load() -> ConversationStore {
        guard let data = try? Data(contentsOf: fileURL),
              let s = try? JSONDecoder().decode(ConversationStore.self, from: data) else {
            return ConversationStore()
        }
        return s
    }

    func save() {
        if let data = try? JSONEncoder().encode(self) { try? data.write(to: ConversationStore.fileURL) }
    }
}

File: windows/Flubber.Core/Conversations.cs
using System.Text.Json;
using Flubber.Core.Platform;

namespace Flubber.Core;

// ============================================================================
// Multiple conversations with the slime, persisted. Port of Conversations.swift.
// ============================================================================

public sealed class Msg
{
    public string Role { get; set; } = "user";   // "user" | "assistant"
    public string Content { get; set; } = "";
    public string? ImagePath { get; set; }        // attached screenshot (clickable thumbnail)
    public string? FilePath { get; set; }         // clickable attached file (.txt transcript)
}

public sealed class Conversation
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Title { get; set; } = "Nueva conversación";
    public List<Msg> Messages { get; set; } = new();

    public static Conversation New() => new()
    {
        Id = Guid.NewGuid().ToString(),
        Title = "Nueva conversación",
        Messages = new(),
    };
}

public sealed class ConversationStore
{
    public List<Conversation> Conversations { get; set; } = new();

    public static ConversationStore Load()
    {
        try
        {
            if (System.IO.File.Exists(Paths.ConversationsJson))
            {
                var json = System.IO.File.ReadAllText(Paths.ConversationsJson);
                var s = JsonSerializer.Deserialize<ConversationStore>(json);
                if (s != null) return s;
            }
        }
        catch { /* ignore */ }
        return new ConversationStore();
    }

    public void Save()
    {
        try { System.IO.File.WriteAllText(Paths.ConversationsJson, JsonSerializer.Serialize(this)); }
        catch { /* ignore */ }
    }
}
